Change representation of debug-commands to a set of command names

yuhan0 opened the pull request:
Note: This is a breaking change that sends debug commands to the client as an unsorted
list of command names without key mapping information, as a means of separating
concerns.


Quick draft of the changes suggested in https://github.com/clojure-emacs/cider/pull/2731
If everything looks alright I'll go ahead with adding changelog notes and tests

Before submitting a PR make sure the following things have been done:

- [x] The commits are consistent with our [contribution guidelines](CONTRIBUTING.md)
- [ ] You've added tests to cover your change(s)
- [x] All tests are passing
- [x] The new code is not generating reflection warnings
- [ ] You've updated the README (if adding/changing middleware)

bbatsov commented on src/cider/nrepl/middleware/debug.clj:
```diff
@@ -297,19 +297,21 @@ this map (identified by a key), and will `dissoc` it afterwards."}
-   "t" :trace})
+  "An unsorted set of commands supported by the debugger."
+  #{:continue
+    :Continue
```
We should use a proper name here - the current one was just to convey the relationship with `C`.

yuhan0 replied:
`:continue-all` and `:inspect-prompt`? Or maybe `:inspect-current` for `p` 
Would that also call for a command name -> display name mapping on the client side? Right now it's just using the plain command names as overlay titles.

bbatsov replied:
You read my mind. I was just writing about some display name mapping on the other ticket. :-)

> :continue-all

👍 

For inspect - let's have the default command inspect the current value and the other to be `inspect-prompt`.

bbatsov commented on src/cider/nrepl/middleware/debug.clj:
```diff
@@ -297,19 +297,21 @@ this map (identified by a key), and will `dissoc` it afterwards."}
+    :next
+    :out
+    :inspect
+    :insPect
```
Same here.

bbatsov commented:
Btw, I guess the second inspect should go in a different PR, as we don't really have that command available currently.

yuhan0 commented:
Ok, I rebased the conflicts, removed the references to :insPect and renamed :Continue to :continue-all. 

Somehow I can't seem to get the test suite working locally - `make test` reports an unrelated error in the info-test and passes the cider.nrepl.debug tests.

```
FAIL in (integration-test) (info_test.clj:161)
info op get info of a java method
expected: (.startsWith (:javadoc response) "https://docs.oracle.com/")
  actual: false
```

yuhan0 commented:
Ok it was a fairly simple change, now all existing tests are passing :)

bbatsov commented:
@yuhan0 Just squash the last two commits and we're good to go.

bbatsov commented:
Great work! Thanks for tackling this! 🙇
